### PR TITLE
Remove app.kubernetes.io/version label from label selectors

### LINKED ISSUE #1694: Should remove app.kubernetes.io/version label from label selectors
### Report

When upgrade from KEDA v2.0.0 to v2.2.0, we encountered the following issue

Deployment.apps "keda-operator" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"keda-operator", "app.kubernetes.io/part-of":"keda-operator", "app.kubernetes.io/version":"2.2.0"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable

### Expected Behavior

We should be able to upgrade KEDA without any issue

### Actual Behavior

Deployment.apps "keda-operator" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"app":"keda-operator", "app.kubernetes.io/part-of":"keda-operator", "app.kubernetes.io/version":"2.2.0"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable

### Steps to Reproduce the Problem

1. Install KEDA by https://github.com/kedacore/keda/releases/download/v2.0.0/keda-2.0.0.yaml
2. upgrade KEDA with https://github.com/kedacore/keda/releases/download/v2.2.0/keda-2.2.0.yaml
3.

### Logs from KEDA operator

```
example
```

### KEDA Version

2.1.0

### Kubernetes Version

1.18

### Platform

Any

### Scaler Details

_No response_

### PR DESCRIPTION
**Pull request by anggao**
_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [ ] Tests have been added
- [ ] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [ ] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [x] Changelog has been updated

Fixes #1694

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -122,8 +122,8 @@ deploy: manifests kustomize
 	$(KUSTOMIZE) edit set image ghcr.io/kedacore/keda=${IMAGE_CONTROLLER}
 	cd config/metrics-server && \
     $(KUSTOMIZE) edit set image ghcr.io/kedacore/keda-metrics-apiserver=${IMAGE_ADAPTER}
-	cd config/default && \
-    $(KUSTOMIZE) edit add label -f app.kubernetes.io/version:${VERSION}
+	@sed -i".out" -e 's@version:[ ].*@version: $(VERSION)@g' config/default/kustomize-config/metadataLabelTransformer.yaml
```
Comment (zroubalik) [inline Makefile:125]: Could you please add comment on this Makefile target to explain why do we need to do this workaround, ie. link this issue in it: https://github.com/kubernetes-sigs/kustomize/issues/1009

File: config/default/kustomization.yaml [hunk 2]
```diff
@@ -19,8 +19,8 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 commonLabels:
   app.kubernetes.io/part-of: keda-operator
-  app.kubernetes.io/version: main
-
+transformers:
```
Comment (zroubalik) [inline config/default/kustomization.yaml:22]:
```suggestion
# Need this transformer to mitigate a problem with inserting labels into selectors,
until this issue is solved: https://github.com/kubernetes-sigs/kustomize/issues/1009
transformers:
```

**Review by zroubalik — COMMENTED** [review]
Looking good, thanks.

Just need to add comments for this change

File: Makefile (hunk 1 above)
Comment (anggao) [reply Makefile:129]: sure comments added.

File: config/default/kustomization.yaml (hunk 2 above)
Comment (anggao) [reply config/default/kustomization.yaml:24]: Yes, fixed.

**Review by zroubalik — APPROVED** [review]
LGTM, thanks!

**Pull request merged**